### PR TITLE
Fix broken regex and add support for discarded packets in get_interfaces_counters

### PR DESCRIPTION
**Pull request by jorgenbl**
The proposed fix will allow for the second match group to be correct (rx_octets and tx_octets)
I also added support for discarded/dropped packets with the "show interface status" command.

### CODE REVIEW

**Comment by mirceaulinic** [comment]
Hi @jorgenbl - thanks for this PR!

The tests are not passing (https://travis-ci.org/napalm-automation/napalm-ios/jobs/206914373) as you need to add a new mock file called `show_interface_summary.txt` under `test/unit/mocked_data/test_get_interfaces_counters/normal` having the output of `show interface summary`. Then you should be ready to go!

**Comment by jorgenbl** [comment]
Hi,

I was testing on gns3 on a 7200 series router, but It seems like you guys are testing on a csr1000v built with vagrant. Will I have to create the same device with vagrant and run the commands to produce the mock file? Can I use the device I have or how does this work? :)

Sorry for the noob questions, but I'm a bit new to this testing procedure.

**Comment by dbarrosop** [comment]
@jorgenbl mocked data can come from anywhere as long as it's from a modern IOS-XE device, they don't really have to come from the same source.

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -1194,11 +1194,13 @@ def get_interfaces_counters(self):
         'rx_broadcast_packets': int,
 
         Currently doesn't determine output broadcasts, multicasts
-        Doesn't determine tx_discards or rx_discards
+        Doesn't determine tx_discards or rx_discards -> fixed with show interface summary
```
Comment (mirceaulinic) [inline napalm_ios/ios.py]: If that's fixed, we can just removed the line.

**Review by mirceaulinic — APPROVED** [review]
This looks good to me.

File: napalm_ios/ios.py [hunk 2]
```diff
@@ -1223,7 +1225,7 @@ def get_interfaces_counters(self):
             for line in interface_str.splitlines():
                 if 'packets input' in line:
                     # '0 packets input, 0 bytes, 0 no buffer'
-                    match = re.search(r"(\d+) packets input.*(\d+) bytes", line)
+                    match = re.search(r"(\d+) packets input.* (\d+) bytes", line)
```
Comment (ktbyers) [inline napalm_ios/ios.py:1227]: Can you expand on what was broken here...both the old and the new RegEx would behave the same way?

File: napalm_ios/ios.py [hunk 3]
```diff
@@ -1258,6 +1260,15 @@ def get_interfaces_counters(self):
                     match = re.search(r"(\d+) output errors", line)
                     counters[interface]['tx_errors'] = int(match.group(1))
                     counters[interface]['tx_discards'] = -1
+            for line in sh_int_sum_cmd_out.splitlines():
+                if interface in line:
+                    regex = r"\b" + interface +\
+                        r"\b\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)" + \
+                        r"\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)"
```
Comment (ktbyers) [inline napalm_ios/ios.py]: You will need to document this regex. I recommend you name the regex groups and put an example of the data pattern above the regex (like: 

```
  Interface                   IHQ       IQD       OHQ       OQD      RXBS      RXPS      TXBS      TXPS      TRTL
-----------------------------------------------------------------------------------------------------------------
  FastEthernet0                 0         0         0         0         0         0         0         0         0
```

**Comment by ktbyers** [comment]
I added one additional improvement that needs made (documenting the regex) and I had a question that I included in the review.

**Comment by mirceaulinic** [comment]
Hi @ktbyers - I think the comments are not visible until the review is finished with explicit Approve/Request Changes/Just comment. I've also missed that many times and posted the comments only few weeks after...

**Comment by ktbyers** [comment]
@mirceaulinic Okay, that comment aspect is annoying...

File: napalm_ios/ios.py (hunk 2 above)
Comment (jorgenbl) [reply napalm_ios/ios.py:1227]: @ktbyers:

Without space it did not seem to catch all of the bytes:
[https://regex101.com/r/cQBAt5/1
](https://regex101.com/r/cQBAt5/1
)
With the space it looks like it works:
[https://regex101.com/r/TZIOf3/1](https://regex101.com/r/TZIOf3/1)

File: napalm_ios/ios.py (hunk 3 above)
Comment (jorgenbl) [reply napalm_ios/ios.py]: Agree, and good idea with the naming of the groups. Did not think about that! 👍

File: napalm_ios/ios.py (hunk 2 above)
Comment (ktbyers) [reply napalm_ios/ios.py:1227]: Okay, makes sense...damn greedy behavior :-)

**Comment by ktbyers** [comment]
I will try to merge this in later today.

**Pull request merged**